flink-clients/src/test/java/org/apache/flink/client/CliFrontendLoggingTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.client;
+
+import org.apache.flink.configuration.ConfigConstants;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.runtime.StreamingMode;
+import org.apache.flink.runtime.minicluster.LocalFlinkMiniCluster;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+
+import static org.apache.flink.client.CliFrontendTestUtils.getTestJarPath;
+import static org.apache.flink.client.CliFrontendTestUtils.getConfigDir;
+import static org.junit.Assert.fail;
+
+public class CliFrontendLoggingTest {
+
+	private static LocalFlinkMiniCluster cluster;
+	private static Configuration config;
+	private static String hostPort;
+	private ByteArrayOutputStream stream = new ByteArrayOutputStream();
+	private CliFrontend cli;
+
+	@Before
+	public void setUp() throws Exception {
+		stream.reset();
+		System.setOut(new PrintStream(stream));
```

Comment:
You are overwriting stdout permanently here. You could save `System.out` to a variable here and set it again in the `@After` method.